### DISCUSSION TITLE
R Special Interest Group on Econometrics?

### DISCUSSION THREAD

**Discussion by pachadotdev** [ropensci/statistical-software-review-book#61] [General]
Hi

IMO it makes sense to create such group. Until today, I've been very hesitant to use statistical packages besides stats:: and h2o::, but I use a lot of data.table and dplyr to clean/reshape data.

Here's a case where I found that FE estimation in R gives different results with different packages
[Three or more FEs: fixest vs lfe vs base vs Stata (reghdfe)](https://github.com/lrberge/fixest/issues/286)

The need to use R packages here is quite clear, the glm function with many fixed effects takes around 100 times the Stata fitting time, and fixest reports the same times as Stata.

While I have created the eflm package with some tricks to speed up the computation, this approach also breaks the robustness in some cases, and at the same time it's the only package I know that tries to be as similar to stats:: outputs.

**Comment by zauster** [comment]
I am interested and willing to participate (I used econometrics in R mostly with gravity models), but I am not sure what exactly the objective of the group would be?
To investigate/monitor the equivalence of results across R packages?

**Comment by mpadge** [comment]
Thanks for enquiring @pachadotdev. The entire endeavour of statistical software review aims to be as multi- and cross-disciplinary as possible, and standards as currently written should definitely encompass the kinds of concerns you raise here. For example, many aspects of the [Standards for *Regression Software*](https://stats-devguide.ropensci.org/standards.html#standards-regression) would ensure appropriate usage, implementation, and documentation of fixed effects estimation. Relevant standards would include:

> Regression Software should explicitly describe the following aspects, and ideally provide extended documentation including summary graphical reports of:

>    RE5.0 *Scaling relationships between sizes of input data (numbers of observations, with potential extension to numbers of variables/columns) and speed of algorithm.*

along with a lot of the [*General Standards for Testing*](https://stats-devguide.ropensci.org/standards.html#testing), which should ensure that algorithmic decisions of the kinds you highlight here would be clearly documented and explicitly tested.

---

On the separate issue of a "Special Interest Group", you could delve into the discussions surrounding an "R-Spatial Working Group" at https://github.com/r-spatial/discuss/issues/37, which has some very detailed considerations behing starting, organising, and funding these kinds of groups.

**Comment by pachadotdev** [comment]
Hi @zauster @mpadge 
I've been working in a port of fixest thinking about the long term. New Mac laptops don't "understand" OMP, so I'm adapting the existing Rcpp codes to a refactored version using cpp11. https://github.com/pachadotdev/fixest/issues/18
All feedback and help is super valuable!
